pkg/summary/summarizers.go

```diff
@@ -218,28 +289,47 @@ func checkStandard(obj data.Object, _ []Condition, summary Summary) Summary {
 	return summary
 }
 
-func checkErrors(_ data.Object, conditions []Condition, summary Summary) Summary {
-	for _, c := range conditions {
-		if (ErrorFalse[c.Type()] && c.Status() == "False") || c.Reason() == "Error" {
-			summary.Error = true
-			summary.Message = append(summary.Message, c.Message())
-			if summary.State == "active" || summary.State == "" {
-				summary.State = "error"
-			}
-			break
-		}
+func checkErrors(data data.Object, conditions []Condition, summary Summary) Summary {
+	conditionMapping := FallbackConditionErrorMapping
+
+	if len(conditions) == 0 {
+		return summary
 	}
 
-	if summary.Error {
+	obj, err := json.Marshal(data)
+	if err != nil {
+		logrus.Debugln("Failed to parse the object: ", err.Error())
 		return summary
 	}
 
+	gvk, detected, err := gvk.Detect(obj)
+	if err != nil {
+		logrus.Debugln("checkErrors: failed to detect GVK: ", err.Error())
+	}
+
+	if detected {
+		gvkConditionMapping, found := GVKConditionErrorMapping[gvk]
+		if found {
+			conditionMapping = gvkConditionMapping
+		}
+	}
```

Comment:
You could simplify this using `GVKConditionErrorMapping[gvk]` if found, and GVKConditionErrorMapping[schema.GroupVersionKind{}] for the fallback if not found.